### PR TITLE
SE-3: Fetch project from github API

### PR DESCRIPTION
**Pull request by phhytrg**
## Purpose
2.a. Display the project structure from github to site
2.b. Display the raw content of each file when clicking on it
- Storing auth state in local storage

### CODE REVIEW

File: backend/src/main/java/com/selflearn/backend/user/UserController.java [hunk 1]
```diff
@@ -1,4 +1,21 @@
...
+import com.selflearn.backend.user.models.User;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/users")
```
Comment (honho001) [inline backend/src/main/java/com/selflearn/backend/user/UserController.java]: consider add version to the url path
```suggestion
@RequestMapping("/v1/users")
```

File: frontend/src/shared/constants/github.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export const GITHUB_OWNER = import.meta.env.VITE_GITHUB_OWNER;
```
Comment (honho001) [inline frontend/src/shared/constants/github.ts]: add these variables to the env.example file

File: frontend/src/project/ProjectTab.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    return nodes.map((node) => {
+      return (
+        <div key={node.path} className="px-2">
+          <Button
+            icon={node.type === 'tree' ? <FolderFilled /> : <FileOutlined />}
+            onClick={async (
+              e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+            ) => {
+              e.preventDefault();
+              if (node.type === 'blob') {
```
Comment (sonhmle) [inline frontend/src/project/ProjectTab.tsx]: put 'blob' string to enum

**Review by sonhmle — COMMENTED** [review]
capture the frontend to review the result (gif or image)
implement github API from the backend instead of using js octokit from the frontend

File: frontend/src/project/ProjectTab.tsx (hunk 3 above)
Comment (phhytrg) [reply frontend/src/project/ProjectTab.tsx]: image

File: frontend/src/main.tsx [hunk 4]
```diff
@@ -31,8 +34,10 @@ const router = createBrowserRouter([
 
 ReactDOM.createRoot(document.getElementById('root')!).render(
   <React.StrictMode>
-    <AuthProvider>
-      <RouterProvider router={router}></RouterProvider>
-    </AuthProvider>
+    <QueryClientProvider client={queryClient}>
+      <AuthProvider>
+        <RouterProvider router={router}></RouterProvider>
```
Comment (sonhmle) [inline frontend/src/main.tsx]:
```suggestion
        <RouterProvider router={router} />
```

File: frontend/src/shared/utils/tree.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { TreeNode } from '../interfaces/tree';
+
+export const buildTreeFromArray = (nodes?: any[]): TreeNode[] => {
```
Comment (sonhmle) [inline frontend/src/shared/utils/tree.ts]: not a must but consider using another type instead of any

**Review by sonhmle — APPROVED** [review]
Nice coding

**Comment by sonhmle** [comment]
put the image in outer comment like this in later PR 😄  @phhytrg 
image

**Pull request merged**